testing/testcafe/tests/editors/overlays/popup.ts

```diff
@@ -100,10 +101,57 @@ test('Popup wrapper left top corner should be the same as the container right le
   }, false, '#innerContainer');
 });
 
-test('There should not be any errors when position.of is html (T946851)', async () => {
-  await t
-    .expect(true).ok();
+test('There should not be any errors when position.of is html (T946851)', async (t) => {
+  await t.expect(true).ok();
 }).before(async () => createWidget('dxPopup', {
   position: { of: 'html' },
   visible: true,
 }));
+
+test('Popup should be centered regarding the window', async (t) => {
+  const popup = new Popup('#container');
+  const initialRect: { bottom: number;
+    top: number;
+    left: number;
+    right: number;
+  } = {
+    bottom: 0,
+    top: 0,
+    left: 0,
+    right: 0,
+  };
+  const wrapperRect = initialRect;
+  const contentRect = initialRect;
+
+  await asyncForEach(['bottom', 'left', 'right', 'top'], async (prop) => {
+    wrapperRect[prop] = await popup
+      .getWrapper()
+      .getBoundingClientRectProperty(prop);
+    contentRect[prop] = await popup.getContent()
+      .getBoundingClientRectProperty(prop);
+  });
+
+  const wrapperVerticalCenter = (wrapperRect.bottom + wrapperRect.top) / 2;
+  const wrapperHorizontalCenter = (wrapperRect.left + wrapperRect.right) / 2;
+  const contentVerticalCenter = (contentRect.bottom + contentRect.top) / 2;
+  const contentHorizontalCenter = (contentRect.left + contentRect.right) / 2;
+
+  await t
+    .expect(wrapperVerticalCenter)
+    .within(contentVerticalCenter - 0.5, contentVerticalCenter + 0.5);
+
+  await t
+    .expect(wrapperHorizontalCenter)
+    .within(contentHorizontalCenter - 0.5, contentHorizontalCenter + 0.5);
+}).before(async () => createWidget('dxPopup', {
+  width: 300,
+  height: 200,
+  visible: true,
+  animation: undefined,
```

Comment:
fx animation is turn off in testcafe test environment. It seems excess

Reply:
if I don't delete the animation or pause for 500 milliseconds, then the content block will give values 10 pixels smaller than it is

Reply:
Nope, it's ok, we will return animation for this file